File: tests/runtime/strings-simple/runner.go
package runner

import (
	"context"
	"fmt"

	wrpc "wrpc.io/go"

	"driver/runner/cat"
)

func Run(ctx context.Context, c wrpc.Invoker) error {
	if err := cat.Foo(ctx, c, "hello"); err != nil {
		return err
	}
	value, err := cat.Bar(ctx, c)
	if err != nil {
		return err
	}
	if value != "world" {
		panic(fmt.Sprintf("expected `world`; got `%v`", value))
	}
	return nil
}

File: tests/runtime/strings-simple/runner.rs
use crate::client::cat::*;

pub async fn run(
    wrpc: &impl ::wit_bindgen_wrpc::wrpc_transport::Invoke<Context = ()>,
) -> ::wit_bindgen_wrpc::anyhow::Result<()> {
    foo(wrpc, (), "hello").await?;

    let t: String = bar(wrpc, ()).await?;
    assert_eq!(t, "world");
    Ok(())
}
